.devcontainer/Dockerfile

```diff
@@ -0,0 +1,84 @@
+
+# Enable new "BUILDKIT" mode for Docker CLI
+ENV DOCKER_BUILDKIT=1
+# debian needs DEV_CONTAINERS_DIR
+ENV DEV_CONTAINERS_DIR=/usr/local/etc/vscode-dev-containers/ 
+
+# Install needed packages and setup non-root user. Use a separate RUN statement to add your
+# own dependencies. A user of "automatic" attempts to reuse an user ID if one already exists.
+ARG USERNAME=automatic
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+COPY library-scripts/*.sh /tmp/library-scripts/
+COPY init.sh /var/init.sh
+COPY first-run-notice.txt /usr/local/etc/vscode-dev-containers/
+
+# Download Kind Docker image so we can bake it into this Docker image instead of having to download it on startup
+RUN bash /tmp/library-scripts/download-frozen-image-v2.sh /var/kindest-node-image kindest/node:v1.24.0
+RUN apt-get update \
+    && /bin/bash /tmp/library-scripts/common-debian.sh "${INSTALL_ZSH}" "${USERNAME}" "${USER_UID}" "${USER_GID}" "${UPGRADE_PACKAGES}" "true" "true" \
+    # Use Docker script from script library to set things up
+    && /bin/bash /tmp/library-scripts/docker-in-docker-debian.sh "${ENABLE_NONROOT_DOCKER}" "${USERNAME}" "${USE_MOBY}" "${DOCKER_VERSION}" \
+    # Clean up
+    && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* /tmp/library-scripts/ \
+    # Install Make
+    && apt update && apt install build-essential -y && apt install make \
+    # Kubectl autocomplete
+    && sudo apt install bash-completion
+
+### ======= ARC dependencies =======
+# Go
+RUN cd /tmp && wget https://go.dev/dl/go1.19.linux-amd64.tar.gz && tar -C /usr/local -xzf go1.19.linux-amd64.tar.gz && rm go1.19.linux-amd64.tar.gz
+ENV PATH="${PATH}:/usr/local/go/bin"
+
+# VS Code Go extension dependencies.
+RUN go install github.com/cweill/gotests/gotests@latest \
+&& go install github.com/fatih/gomodifytags@latest \
+&& go install github.com/josharian/impl@latest \
+&& go install github.com/haya14busa/goplay/cmd/goplay@latest \
+&& go install github.com/go-delve/delve/cmd/dlv@latest \
+&& go install honnef.co/go/tools/cmd/staticcheck@latest \
+&& go install golang.org/x/tools/gopls@latest
+
+# Kind
+RUN go install sigs.k8s.io/kind@latest
+# Hardcoded '/root/go/bin/' because ${HOME} resolves to '/root/.local/go/bin'
+ENV PATH="${PATH}:/root/go/bin"
+
+# Kubectl
+RUN cd /tmp && curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl" && sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl \
+# Autocomplete as per https://kubernetes.io/docs/tasks/tools/included/optional-kubectl-configs-bash-linux/
+&& echo 'source <(kubectl completion bash)' >>~/.bashrc \
+&& echo 'source /usr/share/bash-completion/bash_completion' >>~/.bashrc \
+&& echo 'alias k=kubectl' >>~/.bashrc \
+&& echo 'complete -o default -F __start_kubectl k' >>~/.bashrc
+
+# Kustomize
+RUN curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"  | bash && mv kustomize /usr/bin
```

Comment:
`kustomize` is now built into `kubectl`, `kubectl kustomize .` is the equivalent to `kustomize build .`. This would imply changes to the Makefile I suppose but it's also one less thing we need to install 🎉  [source](https://kustomize.io/)

Reply:
I went back and checked because I remembered being some drift in the two versions in the past, `kubectl` now ships with [`kustomize/v4.5.7`](https://github.com/kubernetes-sigs/kustomize/releases/tag/kustomize%2Fv4.5.7) (source: `kubectl version`). If we're ok changing some commands to compile manifests we can skip this altogether 🎉

Reply:
How do you feel about having an alias until the change is ready. Like `alias kustomize="kubectl kustomize"` to avoid installing it. But great point!

Reply:
Ah, nevermind. We are [using kustomize subcommands](https://github.com/TingluoHuang/actions-runner-controller/blob/master/Makefile#L86) (not included in the one that ships with kubectl), we can likely work around that but it looks more complex than just aliasing 😕 starts to sound a bit out of scope for this maybe?

Reply:
Think so too 👍 
Won't fix here